src/schemas/users.ts

```diff
@@ -0,0 +1,6 @@
+import { Type } from '@sinclair/typebox'
+
+export const UpdateCredentialsSchema = Type.Object({
+  currentPassword: Type.String(),
+  newPassword: Type.String()
+})
```

Comment:
```suggestion
const Password = Type.String({
  // at least 1 upper case, 1 lower case, 1 numeric, 1 special characters with minimal 8 characters.
  pattern: "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",
  minLength: 8
})

export const UpdateCredentialsSchema = Type.Object({
  currentPassword: Password,
  newPassword: Password
})
```

Reply:
> const Password = Type.String({
>   // at least 1 upper case, 1 lower case, 1 numeric, 1 special characters with minimal 8 characters.
>   pattern: "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",
>   minLength: 8
> })

Thanks for the suggestion

The minimum length rule is duplicated in both the regex pattern and the JSON Schema definition, i'll the rules from the schema

Reply:
> the pattern defined only match strings that are exactly 4 characters long

No, it matches any string that is 8 characters or longer. See the last part `.{8,}`.
![image](https://github.com/user-attachments/assets/de13dbbe-f4ca-4598-8d64-db9a760ea53b)

> the minimum length rule is duplicated in both the regex pattern and the JSON Schema definition, i'll keep the rules from the schema.

By looking at the pattern, it is very hard to understand how it works and does it limit the length.
The `minLength` is used here for ensure the string is at least 8 characters and faster understanding on it.
I don't knows which one take higher priority to check in `ajv` but more details is better.

Reply:
> "The minLength is used here for ensure the string is at least 8 characters and faster understanding on it."

True, thanks for your answer